### ISSUE TITLE
NOAUTH Authentication required error for Redis with TLS

### ISSUE THREAD

**Issue by Matejprica** [oliver006/redis_exporter#640]
**Describe the problem**
I am running several redis exporters in docker-compose and it works like a charm, except for one instance which has TLS enabled.
We are using self signed certificates.

**What version of redis_exporter are you running?**
Please run `redis_exporter --version` if you're not sure what version you're running.
[ ] 0.3x.x
[x] 1.x.x

**Screenshots**
If applicable, add screenshots to help explain your question.

docker compose part relevant for this question
![Screenshot_1](https://user-images.githubusercontent.com/63255629/168088014-31bf8d0a-d767-4418-b078-84a76d8f5ba3.png)

Error from docker logs
![Screenshot_3](https://user-images.githubusercontent.com/63255629/168077743-1eaf10cd-c94f-491e-b65c-9d7e56a32ade.png)

Exporter endpoint on Prometheus
![Screenshot_4](https://user-images.githubusercontent.com/63255629/168078102-aaf7b1db-e21b-492c-b9fc-46d16aff6c6d.png)

**Additional context**
host, password and path to certificates has been changed ofc. Password and certificates are correct, aswell as path (no typo). I can connect to redis via redis-cli with password and CA cert parameters. But when I pass them to docker compose as environment variables I keep getting the NOAUTH error. I tried also adding TLS_SERVER_CERT_FILE and TLS_SERVER_CERT_KEY aswell as TLS_CA_CERT_FILE instead of TLS_SERVER_CA_CERT_FILE which is in picture above. I tried writing all env vars in key:"value" way aswell as in key=value way (as in picture above). I tried with and without REDIS_EXPORTER_SKIP_TLS_VERIFICATION but no luck. Always the same NOAUTH error. I feel like I am missing something very trivial but I cant figure it out, my biggest suspect is REDIS_EXPORTER_SKIP_TLS_VERIFICATION flag. Did I set it correctly? Anyway if you have any idea or suggestion what could be wrong, I would greatly appreciate it

**Comment by bjosv** [comment]
Since you get NOAUTH from Redis the certificates seem ok to me, and there might be something wrong with the password.
Is there any space in the password that requires it to be put in `"xxx"` or any other non-ascii characters?

**Comment by Matejprica** [comment]
Nope, password consists of numbers and lowercase/uppercase english alphabet letters. The only different char is = which is at the end of the password, but that shouldnt affect it. I've wrapped password in quotes just in case, but still the same error message. Password is correct because I can auth to redis via redis-cli with the same password. There is also no typo in password :) checked it several times, copied and even wrote it char by char

**Comment by Matejprica** [comment]
I finally figured it out when I actually inspected the container... When passing TLS env vars as env var in docker compose, they are not added as actual env var in container. Only password.
![InkedScreenshot_1_LI](https://user-images.githubusercontent.com/63255629/168327648-2dacdf17-3489-4964-8c36-c464732b4e79.jpg)

So after I manually created container with:
docker run -d -p 9119:9121 oliver006/redis_exporter:v1.37.0 --redis.password='xxx' --skip-tls-verification
it finally worked and I can connect to redis with just a password, and skip tls verification flag is added to CMDs
![InkedScreenshot_2_LI](https://user-images.githubusercontent.com/63255629/168328146-5f873fb1-7663-4130-9363-ff8b0e80e13f.jpg)
Not sure is that a bug or working as intended, but posting it if anyone else bumps into the same issue and finds this useful. 
Thanks @oliver006  for awesome repo and @bjosv  for adding TLS support.
Closing this question
